checkrebuild

```diff
@@ -22,17 +22,26 @@ get_unofficial_pkgs() {
     comm -23 <(pacman -Qq | sort) <(pacman -Sql "${repos_to_skip[@]}" 2>/dev/null | sort)
 }
 
-filter_executable() {
-    while read -r file; do
-        [[ -f "$file" && -x "$file" ]] && echo "$file"
-    done
+newline2null(){
+    perl -pe 's/\n/\0/'
 }
 
 get_package_files() {
     xargs pacman -Qql |
-    perl -pe 's/\n/\0/' |
-    xargs -0 readlink -f |
-    filter_executable
+    newline2null |
+    xargs -0 readlink -e |
+    sort -u
+}
+
+filter_executable() {
+    newline2null |
+    LANG=C xargs -0 stat --printf "%F %a\t%n\n" |
+    grep -P "^regular file [^64]" |
```

Comment:
More precise is `^regular file [1357]`

Reply:
To be really we should use ` ([0-9]|)[1735][0-9]{2}`
```sh
pacman -Qlq|perl -pe 's/\n/\0/'|xargs -0 readlink -ez|LANG=C xargs -0 stat --printf "%F\t%a\t%n\n"|grep -P "^regular file\t"|cut -f2|sort|uniq -c
      2 0
      5 1755
      8 2755
      2 4110
      1 440
  10450 444
      1 4555
     11 4711
      2 4750
     48 4755
      1 4775
      1 505
    302 555
     42 600
     91 604
     70 640
1202888 644
     20 655
   2957 664
      2 666
      9 6755
     20 700
      2 711
    436 744
      2 750
      3 754
  47802 755
     59 775
      9 77
```
Btw. those two `0` is from `/etc/{g}shadow` :smirk:

Reply:
I see you already changed it to `([1-6]|)[1357]`, what does the first part do exactly, could you clarify please?

By the way:

```
$ stat --printf "%F %a\t%n" /etc/shadow
regular file 600	/etc/shadow                                                                     
$ stat --printf "%F %a\t%n" /etc/gshadow
regular file 600	/etc/gshadow
```

Reply:
First part is for `setuid,setgid,sticki_byt` and is described in `chmod(1)`
Yep, my system has some legacy permissins (continuous update since 2014 :face_with_head_bandage: ).

Reply:
I see 🙂 And how does this regex work exactly, you check if it starts with `[1-6]` or if it starts with `[1357]`? Is it the same as `[1-7]`?

Unless I misunderstand the syntax, I think your earlier suggestion was more correct, I would change it to this perhaps: `\d?[1735]\d\d`

Reply:
Yap, looks nicer.